pallets/bridge_transfer/src/lib.rs

```diff
@@ -66,13 +69,27 @@ decl_module! {
 		// Initiation calls. These start a bridge transfer.
 		//
 
-		/// Transfers an arbitrary hash to a (whitelisted) destination chain.
+		/// Transfers an arbitrary signed bitcoin tx to a (whitelisted) destination chain.
 		#[weight = 195_000_000]
-		pub fn transfer_hash(origin, hash: T::Hash, dest_id: bridge::ChainId) -> DispatchResult {
+		pub fn transfer_lottery(origin, round_id: u32, token_id: u32, payload: Vec<u8>, dest_id: bridge::ChainId) -> DispatchResult {
 			ensure_signed(origin)?;
 
-			let resource_id = Self::bridge_hashid();
-			let metadata: Vec<u8> = hash.as_ref().to_vec();
+			let resource_id = Self::bridge_lotteryid();
+
+			let payload_len: u32 = payload.len().saturated_into();
+			let mut encoded_payload_len: Vec<u8> = payload_len.encode();
+			encoded_payload_len.reverse();
+
+			let mut encoded_round_id: Vec<u8> = round_id.encode();
+			encoded_round_id.reverse();
+
+			let mut encoded_token_id: Vec<u8> = token_id.encode();
+			encoded_token_id.reverse();
```

Comment:
We can pacak them all, and serialize by parity-scale-codec. This data will only be read by our js clients (decode, and broadcast to BTC blockchain). We don't need to make it compatible with Ethereum ABI.

So I'd suggest to write something like `let matadata: Vec<_> = (round_id, token_id, payload).encode()`